### ISSUE TITLE
Migration from SQLITE to MYSQL

### ISSUE THREAD

**Issue by all3kcis** [pagermon/pagermon#528]
**Is your feature request related to a problem? Please describe.**
I want to migrate my actual DB (SQLITE) approx 130k row in messages to MYSQL DB, but all the tools I have tested report errors

**Describe the solution you'd like**
I didn't find it in the wiki but maybe you have something planned for that already?

**Comment by DanrwAU** [comment]
Export to tsv from sqlite then import into mysql is how I do this.

**Comment by all3kcis** [comment]
I exported in CSV, I met problems with the field alias_id of the table message, in CSV the column was empty and MYSQL waited a value "NULL" in this case, I could manage by modifying these lines with Excel.

To recreate the tables I configured pagermon and then restarted it. But it doesn't recreate the "messages_search..." tables, is that a problem?

**Comment by eopo** [comment]
The best way would be to change config to mysql and then run pagermon. This will start the database initialization process. After that, shut down pagermon, export your SQLITE-tables for messages and aliases to SQL insert statements. First import your aliases, then your messages. If you try it the other way around you will get reference errors.

**Comment by BuildTheRobots** [comment]
FYI: I found exporting from SQLITE to TSV was causing the data to get corrupted when opening in LibraOffice.
I ended up exporting using a custom separation character (®) and then writing a tiny perl script to convert each line to a CURL command which uses the API to insert the messages into the new database, which seems to be taking care of aliases and search indexes for me, without me needing to understand the SQL data structure.
180,000 messages; it's been going since last night and seems to be working as expected; just somewhat slow.
